src/json_path.rs

```diff
@@ -913,32 +922,33 @@ impl<'i, UPTG: UserPathTrackerGenerator> PathCalculator<'i, UPTG> {
                         self.calc_range(pairs, curr, json, path_tracker, calc_data);
                     }
                     Rule::filter => {
-                        if flat_arrays_on_filter && json.get_type() == SelectValueType::Array {
+                        if json.get_type() == SelectValueType::Array
```

Comment:
Is there any impact on performance from this change? (always flattening also without filter)

Reply:
Not that I can think of...